src/wgpu/wgpu-shader-object.cpp

```diff
@@ -402,7 +402,8 @@ Result ShaderObjectImpl::_ensureOrdinaryDataBufferCreatedIfNeeded(
         bufferDesc.defaultState = ResourceState::ConstantBuffer;
         bufferDesc.memoryType = MemoryType::DeviceLocal;
         ComPtr<IBuffer> buffer;
-        SLANG_RETURN_ON_FAIL(encoder->m_device->createBuffer(bufferDesc, nullptr, buffer.writeRef()));
+        void * initData = m_data.getBuffer();
+        SLANG_RETURN_ON_FAIL(encoder->m_device->createBuffer(bufferDesc, initData, buffer.writeRef()));
```

Comment:
```suggestion
        SLANG_RETURN_ON_FAIL(encoder->m_device->createBuffer(bufferDesc, m_constantBufferData.data(), buffer.writeRef()));
```

Reply:
Done except I check for empty as well.

Reply:
that's not necessary as we exit early if there is no constant data .. see at the start of the function
```c++
    if (m_constantBufferSize == 0)
    {
        return SLANG_OK;
    }
```

Reply:
Done.